cdc/sink/dmlsink/txn/worker.go

```diff
@@ -122,27 +117,39 @@ func (w *worker) runLoop() error {
 				zap.Int("workerID", w.ID))
 			return nil
 		case txn := <-w.txnCh.Out():
+			// we get the data from txnCh.out until no more data here or reached the state that can be flushed.
+			// If no more daa in txnCh.out, and also not reached the state that can be flushed,
+			// we will wait for 10ms and then do flush to avoid too much flush wish small amount of txns.
 			if txn.txnEvent != nil {
 				needFlush = w.onEvent(txn)
+				if !needFlush {
```

Comment:
It seems that the core idea here is to ensure that the flush interval is greater than 10ms.  Maybe we could record lastFlushTime at the end of each flush and check it in each ticker?

Reply:
I believe both of these two methods are feasible, and I'm wondering if the current code might be a bit more straightforward and easier to understand?

Reply:
In general, the nested logic is more complex.

Reply:
I think it's ok here orz